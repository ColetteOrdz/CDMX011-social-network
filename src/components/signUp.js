/* eslint-disable */
import { onNavigate } from '../routes.js';
import { register } from '../lib/fireBase.js';

export const toViewSignUp = (container) => {
    console.log('Estoy en signUp');
    const html = ` <section class='container signup-form'>
    <img class='logo' src='img/PIC&ART.png' alt='logo'>
    <form id='submitForm'>
        <br><label value='email'>E-mail*</label>
        <input id='singUpEmail' type='text' placeholder='e-mail' required/><br>
        <br><label value='password'>Password*</label>
        <input id='singUpPassword' type='password' placeholder='Password' required/>
        <input id='viewPass' src='../img/eye.png' type='image'/>
        <br><input type='submit' class='btn_log signup' value='SIGN UP' /><br>
        <p class="error"></p>
      
        
       <br>
       <br>
    </form>
    <a href='javascript:history.back();'> <img class='return' src='img/return-logo.png' alt='logo' > Back </a>
</section>`;
    container.innerHTML = html

    document.getElementById('viewPass').addEventListener('click', (e) => {
        e.preventDefault();
        const inputType = document.getElementById('singUpPassword');
        inputType.type == 'password' ? inputType.type = 'text' : inputType.type = 'password';
    });

    const singUpform = document.querySelector('#submitForm');
    singUpform.addEventListener('submit', (e) => {
        const email = document.querySelector('#singUpEmail').value;
        const password = document.querySelector('#singUpPassword').value;

        e.preventDefault();
        e.stopPropagation();
        console.log('estoy evitando el reset');
        console.log(email, password);

        register(email, password)
            .then(() => {
                console.log('then');
                onNavigate('/TimeLine');
            })
            .catch((error) => {
                const alertaError = error.message;
                document.querySelector('.error').innerHTML = `${alertaError}`;
                console.log(error.message);
            });
    });
};