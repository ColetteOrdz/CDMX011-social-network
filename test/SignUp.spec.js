/**
 * @jest-environment jsdom
 */
/* eslint-disable */

import './globals/firebase.js';
import { toViewSignUp } from '../src/components/signUp.js';

describe("ColecciÃ³n de test en SignUp", () => {
    document.body.innerHTML=`<div id="root"></div>`;
    
    it("Should render in SignUp", () => {
        const rootDiv=document.getElementById('root');
        //const component=toViewLogIn();
        toViewSignUp(rootDiv);
        //console.log(rootDiv.innerHTML);
       expect(rootDiv.innerHTML).toMatchSnapshot();
    });

    it("should log when btn signUp is clicked", () =>{
        const mockSignUp= jest.fn();
        mockSignUp.mockImplementation(()=>Promise.resolve()); 

        firebase.auth= jest.fn().mockImplementation(()=>({
            createUserWithEmailAndPassword: mockSignUp,
        }));
        const rootDiv=document.getElementById('root');
        //const component=toViewLogIn();
        toViewSignUp(rootDiv);

        const email='testing@test.com';
        const password='123456.';

        document.getElementById('singUpEmail').value=email;
        document.getElementById('singUpPassword').value=password;

        document.getElementById('btnSignUp').click();
        expect(mockSignUp).toHaveBeenCalledWith(email, password);

    })

    
      });