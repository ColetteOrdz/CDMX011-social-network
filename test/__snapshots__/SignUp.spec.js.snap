// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Colecci√≥n de test en SignUp Should render in SignUp 1`] = `
" <section class=\\"container signup-form\\">
    <img class=\\"logo\\" src=\\"img/PIC&amp;ART.png\\" alt=\\"logo\\">
    <form id=\\"submitForm\\">
        <br><label value=\\"email\\">E-mail*</label>
        <input id=\\"singUpEmail\\" type=\\"text\\" placeholder=\\"e-mail\\" required=\\"\\"><br>
        <br><label value=\\"password\\">Password*</label>
        <input id=\\"singUpPassword\\" type=\\"password\\" placeholder=\\"Password\\" required=\\"\\">
        <input id=\\"viewPass\\" src=\\"../img/eye.png\\" type=\\"image\\">
        <br><input type=\\"submit\\" class=\\"btn_log signup\\" id=\\"btnSignUp\\" value=\\"SIGN UP\\"><br>
        <p class=\\"error\\"></p> 
        <br>
        <br>
    </form>
    <a href=\\"javascript:history.back();\\"> <img class=\\"return\\" src=\\"img/return-logo.png\\" alt=\\"logo\\"> Back </a>
</section>"
`;
